services/agents/registry.go

```diff
@@ -62,9 +62,9 @@ type pmmAgentInfo struct {
 //
 // TODO Split into several types https://jira.percona.com/browse/PMM-4932
 type Registry struct {
-	db         *reform.DB
-	prometheus prometheusService
-	qanClient  qanClient
+	db             *reform.DB
+	scrapeServices []prometheusService
```

Comment:
isn't it better to create some proxy service which will pass request to prometheus service or victoria metrics service or both based on environment variables?

Reply:
And then remove prometheus will be easier.

Reply:
we don't need to do orchestration for removing of prometheus, i think this is not a hard work. 
Please, also take into account that `vmalert` will be added and it has similar interface and should be reload by when config changes, as right now, this is one call (change settings) we need to ask every service about config verification/updating